projects/stabble/index.js

```diff
@@ -0,0 +1,256 @@
+const sdk = require("@defillama/sdk");
+const { Program } = require("@project-serum/anchor");
+const { PublicKey } = require("@solana/web3.js");
+const { getProvider, sumTokens2 } = require("../helper/solana");
+
+async function weightedSwapTvl() {
+  const VAULT_ID = "w8edo9a9TDw52c1rBmVbP6dNakaAuFiPjDd52ZJwwVi";
+
+  const provider = getProvider();
+  const programId = new PublicKey("swapFpHZwjELNnjvThjajtiVmkz3yPQEHjLtka2fwHW");
+  const program = new Program(WEIGHTED_SWAP_IDL, programId, provider);
+  const pools = await program.account.pool.all([
+    {
+      memcmp: {
+        offset: 40,
+        bytes: VAULT_ID,
+      },
+    },
+  ]);
+  const tokens = Array.from(new Set(pools.map(({ account }) => account.tokens.map(({ mint }) => mint)).flat()));
+  const owner = findVaultAuthorityAddress(new PublicKey(VAULT_ID));
+
+  return sumTokens2({
+    tokens,
+    owner,
+  });
+}
+
+async function stableSwapTvl() {
+  const VAULT_ID = "stab1io8dHvK26KoHmTwwHyYmHRbUWbyEJx6CdrGabC";
+
+  const provider = getProvider();
+  const programId = new PublicKey("swapNyd8XiQwJ6ianp9snpu4brUqFxadzvHebnAXjJZ");
+  const program = new Program(STABLE_SWAP_IDL, programId, provider);
+  const pools = await program.account.pool.all([
+    {
+      memcmp: {
+        offset: 40,
+        bytes: VAULT_ID,
+      },
+    },
+  ]);
+  const tokens = Array.from(new Set(pools.map(({ account }) => account.tokens.map(({ mint }) => mint)).flat()));
+  const owner = findVaultAuthorityAddress(new PublicKey(VAULT_ID));
+
+  return sumTokens2({
+    tokens,
+    owner,
+  });
+}
+
+module.exports = {
+  timetravel: false,
+  solana: { tvl: sdk.util.sumChainTvls([weightedSwapTvl, stableSwapTvl]) },
+};
+
+function findVaultAuthorityAddress(vaultKey) {
+  return PublicKey.findProgramAddressSync(
+    [Buffer.from("vault_authority"), vaultKey.toBuffer()],
+    new PublicKey("vo1tWgqZMjG61Z2T9qUaMYKqZ75CYzMuaZ2LZP1n7HV")
+  )[0];
+}
+
+const WEIGHTED_SWAP_IDL = {
```

Comment:
@g1nt0ki I've defined anchor 0.26.0 compatible IDL manually as our contract is written in anchor 0.30.0.